### PR TITLE
github: Summer cleaning

### PR DESCRIPTION
**Pull request by Morriar**
Some cleaning to the `github` lib.

Most interesting changes:

* kill `github_curl` and migrate clients (`github_merge`, `github_search_for_jni`)
* clean deserialization process heuristics for something more robust and readable (but a bit slower)
* array pagination
* kill `GithubEntity`
* remove ISO date methods
* mock testing for `github::wallet`
* `wallet` uses a `Logger`

This PR can be read commit by commit.

### CODE REVIEW

File: lib/github/api.nit [hunk 1]
```diff
@@ -19,7 +19,6 @@
 # For most use-cases you need to use the `GithubAPI` client.
 module api
 
-# TODO to remove
```
Comment (privat) [inline lib/github/api.nit]: but but but you add it 14 commits ago https://github.com/nitlang/nit/pull/2765/commits/ae3ec8baee7660fb029a9be55c5a20fca7cc6fc5#diff-910b30e850bf6fd2777c60a2a1627a07R22

File: contrib/github_merge/github_merge.nit [hunk 2]
```diff
@@ -157,23 +153,24 @@ end
...
+	mergemsg.add "Merge: {pull.title}\n\n"
+	mergemsg.add "{pull.body or else "N/A"}\n\n"
+	mergemsg.add "Pull-Request: #{pull.number}\n"
+	for rev in revs do
+		mergemsg.add "Reviewed-by: {rev}\n"
 	end
 	mergemsg.write_to_file("mergemsg")
 
-	var sha = pr["head"].json_as_map["sha"].as(String)
+	var sha = pull.head.sha
```
Comment (privat) [inline contrib/github_merge/github_merge.nit]: so much grace

File: lib/github/api.nit [hunk 3]
```diff
@@ -1122,6 +1131,10 @@ class GithubDeserializer
 			return "Branch"
 		else if raw.has_key("total_count") and raw.has_key("items") then
 			return "SearchResults"
+		else if raw.has_key("total") and raw.has_key("weeks") then
+			return "ContributorStats"
+		else if raw.has_key("a") and raw.has_key("d") and raw.has_key("c") then
```
Comment (privat) [inline lib/github/api.nit]: I could have named those fields!

File: lib/github/cache.nit [hunk 4]
```diff
@@ -58,7 +58,7 @@ redef class GithubAPI
 	redef fun get(key, headers, data) do
 		if not enable_cache then return super
 		if store.has_key(key) then
-			message(1, "Get {key} (cache)")
+			# print "Get {key} (cache)" # debug
```
Comment (privat) [inline lib/github/cache.nit]: Seriously ? Replacing old logging infrastructure with greasy commented hard coded prints. I like that!

File: lib/github/api.nit (hunk 1 above)
Comment (Morriar) [reply lib/github/api.nit]: Yes but it's only used for a hack in the deserialization process using `path`:

https://github.com/nitlang/nit/pull/2765/files#diff-910b30e850bf6fd2777c60a2a1627a07R1134

File: contrib/github_merge/github_merge.nit (hunk 2 above)
Comment (Morriar) [reply contrib/github_merge/github_merge.nit]: The previous version was burning my eyes.

File: lib/github/api.nit (hunk 3 above)
Comment (Morriar) [reply lib/github/api.nit]: Actually, there is a striking resemblance between you and Tenderlove... At least for the bad jokes: p

https://twitter.com/tenderlove/status/714540289675763712?lang=en

File: lib/github/cache.nit (hunk 4 above)
Comment (Morriar) [reply lib/github/cache.nit]: Next step is to kill `json_store`.

If I hit it hard enough, it should take `github_cache` down with it.

**Comment by Morriar** [comment]
> I did read it ! 🏅

Congratz! :trophy:

**Pull request merged**